src/common-utils/src/url.ts

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+// Copyright contributors to the kepler.gl project
+
+/**
+ * Allows to break down a url into multiple params
+ * from http://blog.stevenlevithan.com/archives/parseuri
+ */
+export function parseUri(str: string): {[key: string]: any} {
```

Comment:
Nit: This looks like rather hacky old code. Use standard `URL` API that is available both in browser and Node.js?  https://nodejs.org/docs/v22.12.0/api/url.html#url